tests/unittests/cmd/test_status.py

```diff
@@ -1005,13 +999,70 @@ def test_retry_no_wait(self, mocker, capsys):
         )
         mocker.patch("time.time", side_effect=[1, 2, 50])
         assert (
-            _get_error_or_running_from_systemd_with_retry(
-                UXAppStatus.ERROR, wait=False
-            )
+            _get_error_or_running_from_systemd(UXAppStatus.ERROR, wait=False)
             is None
         )
         assert 1 == m_subp.call_count
         assert (
             "Failed to get status from systemd. "
             "Cloud-init status may be inaccurate."
         ) in capsys.readouterr().err
+
+
+class TestQuerySystemctl:
+    def test_query_systemctl(self, mocker):
+        m_subp = mocker.patch(
+            f"{M_PATH}subp.subp",
+            return_value=SubpResult(stdout="hello", stderr=None),
+        )
+        assert status.query_systemctl(["some", "args"], wait=False) == "hello"
+        m_subp.assert_called_once_with(["systemctl", "some", "args"])
+
+    def test_query_systemctl_with_exception(self, mocker, capsys):
+        m_subp = mocker.patch(
+            f"{M_PATH}subp.subp",
+            side_effect=subp.ProcessExecutionError(
+                "Message recipient disconnected", stderr="oh noes!"
+            ),
+        )
+        assert status.query_systemctl(["some", "args"], wait=False) == ""
+        m_subp.assert_called_once_with(["systemctl", "some", "args"])
+        assert "Error from systemctl: oh noes!" in capsys.readouterr().err
+
+    def test_query_systemctl_wait_with_exception(self, mocker):
+        m_sleep = mocker.patch(f"{M_PATH}sleep")
+        m_subp = mocker.patch(
+            f"{M_PATH}subp.subp",
+            side_effect=[
+                subp.ProcessExecutionError("Message recipient disconnected"),
+                subp.ProcessExecutionError("Message recipient disconnected"),
+                subp.ProcessExecutionError("Message recipient disconnected"),
+                SubpResult(stdout="hello", stderr=None),
+            ],
+        )
+
+        assert status.query_systemctl(["some", "args"], wait=True) == "hello"
+        assert m_subp.call_count == 4
+        assert m_sleep.call_count == 3
+
+    def test_query_systemctl_wait_with_exception_status(self, mocker):
+        m_sleep = mocker.patch(f"{M_PATH}sleep")
+        m_subp = mocker.patch(
+            f"{M_PATH}subp.subp",
+            side_effect=[
+                subp.ProcessExecutionError("Message recipient disconnected"),
+                subp.ProcessExecutionError("Message recipient disconnected"),
+                subp.ProcessExecutionError("Message recipient disconnected"),
+            ],
```

Comment:
Can drop the other side_effects
```suggestion
            side_effect=subp.ProcessExecutionError("Message recipient disconnected"),
```